factory/stateComponents.go

```diff
@@ -0,0 +1,97 @@
+package factory
+
+import (
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	factoryState "github.com/ElrondNetwork/elrond-go/data/state/factory"
+	"github.com/ElrondNetwork/elrond-go/data/trie/factory"
+	"github.com/ElrondNetwork/elrond-go/sharding"
+	"github.com/ElrondNetwork/elrond-go/storage"
+)
+
+// StateComponentsFactoryArgs holds the arguments needed for creating a state components factory
+type StateComponentsFactoryArgs struct {
```

Comment:
This should be merged with the Data component, as stated in TDA

Reply:
I would leave it as a TODO for now and if this PR doesn't have to be merged quickly, I will do it on this PR